src/main/java/com/example/onlinebookshop/services/impl/UserServiceImpl.java

```diff
@@ -23,7 +33,23 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto)
             throw new RegistrationException("User with email "
                     + requestDto.getEmail() + " already exists");
         }
-        User savedUser = userRepository.save(userMapper.toUser(requestDto));
+        requestDto.setPassword(passwordEncoder.encode(requestDto.getPassword()));
```

Comment:
Let create User first
```suggestion
        User user = userMapper.toUser(requestDto);
        user.setRoles( // set USER role
        user.setPassword(passwordEncoder.encode(request.password()));
        userRepository.save(user);
        return userMapper.toUserResponseDto(user);
```